[quant][gpu][improvement] Added eager mode test for quantized cuda resnet

dzdang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #75331

Summary:
A test has been added that compares CUDA and CPU implementations for quantized resnet18

Test plan:
```
python test/test_quantization.py TestQuantizeEagerPTQStaticCUDA.test_resnet18
```

Summary:
work in progress

Differential Revision: [D35788446](https://our.internmc.facebook.com/intern/diff/D35788446)

dzdang committed:
[quant] benchmarking quantized cuda resnet

[ghstack-poisoned]

dzdang committed:
Update on "[quant] benchmarking quantized cuda resnet"




[ghstack-poisoned]

dzdang committed:
Update on "[quant] benchmarking quantized cuda resnet"




[ghstack-poisoned]

dzdang committed:
Update on "[quant] benchmarking quantized cuda resnet"




[ghstack-poisoned]

albanD commented on torch/nn/functional.py:
```diff
@@ -770,7 +770,10 @@ def _max_pool2d(
         stride = torch.jit.annotate(List[int], [])
-    return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)
+    if input.is_quantized and input.is_cuda:
+        return torch.ops.quantized.max_pool2d(input, _pair(kernel_size), _pair(stride), _pair(padding), _pair(dilation), ceil_mode)
```
Why isn't the regular `max_pool2d` already handling this case?
We don't have any quantized special case in this file.

albanD reviewed (changes requested):
Note also that using the non-overload specific version of torch.ops is very slow. So you most likely don't want to do that either.

dzdang committed:
Update on "[quant][gpu][improvement] Added eager mode test for quantized cuda resnet"


Summary:
A test has been added that compares CUDA and CPU implementations for quantized resnet18

Test plan:
```
python test/test_quantization.py TestQuantizeEagerPTQStaticCUDA.test_resnet18
```

Summary:
work in progress

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35788446).

dzdang committed:
Update on "[quant][gpu][improvement] Added eager mode test for quantized cuda resnet"


Summary:
A test has been added that compares CUDA and CPU implementations for quantized resnet18

Test plan:
```
python test/test_quantization.py TestQuantizeEagerPTQStaticCUDA.test_resnet18
```

Summary:
work in progress

Differential Revision: [D35788446](https://our.internmc.facebook.com/intern/diff/D35788446)

[ghstack-poisoned]
